### PR TITLE
[docs] passing LogSoftmax into NLLLoss

### LINKED ISSUE #96795: Docs request: combination of LogSoftmax and NLLLoss in CrossEntropyLoss
### 📚 The doc issue

From https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html

> Note that this case is equivalent to the combination of [`LogSoftmax`](https://pytorch.org/docs/stable/generated/torch.nn.LogSoftmax.html#torch.nn.LogSoftmax) and [`NLLLoss`](https://pytorch.org/docs/stable/generated/torch.nn.NLLLoss.html#torch.nn.NLLLoss).

This requires readers to have to look at each formula, and figure out how to combine them.  It would be better to explicitly state how they are combined.

### Suggest a potential alternative/fix

I think the below wording is equally concise, but more informative:

> Note that this case is equivalent to a [`LogSoftmax`](https://pytorch.org/docs/stable/generated/torch.nn.LogSoftmax.html#torch.nn.LogSoftmax) passed into a [`NLLLoss`](https://pytorch.org/docs/stable/generated/torch.nn.NLLLoss.html#torch.nn.NLLLoss).

If you like this, I can open a PR!

cc @svekars @carljparker @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

### PR DESCRIPTION
**Pull request by jamesbraza**
Fixes https://github.com/pytorch/pytorch/issues/96795

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -1070,8 +1070,8 @@ class CrossEntropyLoss(_WeightedLoss):
                 \text{if reduction} = \text{`sum'.}
             \end{cases}
 
-      Note that this case is equivalent to the combination of :class:`~torch.nn.LogSoftmax` and
-      :class:`~torch.nn.NLLLoss`.
+      Note that this case is equivalent to a :class:`~torch.nn.LogSoftmax`
+      passed into a :class:`~torch.nn.NLLLoss`.
```
Comment (soulitzer) [inline torch/nn/modules/loss.py:1074]: Maybe its slightly clearer if we reword to something like: 
```
Note that this case is equivalent to applying :class:`~torch.nn.LogSoftmax` on input, followed by :class:`~torch.nn.NLLLoss`. 
```

Comment (jamesbraza) [reply torch/nn/modules/loss.py]: Good suggestion!  Implemented it

**Comment by jamesbraza** [comment]
@jbschlosser I didn't actually remove you as a requester, for some reason GitHub did when I re-requested @soulitzer.   I don't have the ability to re-add you though

**Comment by soulitzer** [comment]
Thanks for the update, just need CLA and we should be good to go!

**Comment by jamesbraza** [comment]
Alright, completed CLA.  Dang, PyTorch is legit!

**Review by soulitzer — APPROVED** [review]
